amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -185,9 +188,11 @@ void disconnect() {
         }
 
         // If ble connection is closed, there's no need to keep mqtt connection open.
+        mBleConnectionState = AmazonFreeRTOSConstants.BleConnectionState.BLE_DISCONNECTED;
         if (mMqttConnectionState != AmazonFreeRTOSConstants.MqttConnectionState.MQTT_Disconnected) {
             disconnectFromIot();
         }
+        mBleConnectionStatusCallback.onBleConnectionStatusChanged(mBleConnectionState);
```

Comment:
Will this lead to callback invoked twice for a successful disconnect, other one being from onConnectionStateChange() ?

Reply:
I moved the whole disconnect logic to the onConnectionStateChange() callback.